beetmoverscript/script.py

```diff
@@ -316,26 +336,49 @@ def list_bucket_objects(context, s3_resource, prefix):
         # move beets
         for artifact in artifacts_to_beetmove[locale]:
             source = artifacts_to_beetmove[locale][artifact]
-            artifact_pretty_name = manifest['mapping'][locale][artifact]['s3_key']
 
             # update buildhub.json file
             # if there is no installer then there will be no buildhub.json artifact
             # in logical coding terms, this means that if installer_path is an empty
             # string, then this if-block is never reached
             if artifact == BUILDHUB_ARTIFACT:
-                write_json(source, get_updated_buildhub_artifact(source, installer_path, context, manifest, locale))
-
-            destinations = [os.path.join(manifest["s3_bucket_path"],
-                                         dest) for dest in
-                            manifest['mapping'][locale][artifact]['destinations']]
+                write_json(source, get_updated_buildhub_artifact(
+                    source, installer_path, context, locale, manifest=manifest, artifact_map=artifact_map))
+
+            if artifact_map:
+                task_id = get_taskId_from_full_path(source)
+                # Should only ever be one (taskId, locale) match.
+                map_entry = extract_file_config_from_artifact_map(
+                    artifact_map, artifact, task_id, locale)
+
+                artifact_pretty_name = map_entry['checksums_path']
+                destinations = map_entry['destinations']
+                update_balrog_manifest = map_entry.get('update_balrog_manifest', False)
+                balrog_format = map_entry.get('balrog_format', '')
+                from_buildid = map_entry.get('from_buildid')
+            else:
+                artifact_pretty_name = manifest['mapping'][locale][artifact]['s3_key']
+                destinations = [os.path.join(manifest["s3_bucket_path"],
+                                             dest) for dest in
+                                manifest['mapping'][locale][artifact]['destinations']]
+                update_balrog_manifest = manifest['mapping'][locale][artifact].get(
+                    'update_balrog_manifest')
+                # Adjust old template format.
+                # artifact map specifies these separately.
+                # templates say "update_balrog_manifest": true
+                # or "update_balrog_manifest": {"format": mozinfo}
+                if isinstance(update_balrog_manifest, dict):
```

Comment:
Once declarative artifacts is in place, I'd really like to separate this behavior. I'm not particularly fond of this variable reuse, it makes it complicated to read for someone not familiar with the code. However, this is a carry-over thing and is not blocking existing PR. We'll address it hopefully in H1 2019.

Reply:
Agreed, it's not easy to read, and it should hopefully disappear with the old manifest templates.